### PR TITLE
Topic2

### PR DESCRIPTION
**Pull request by Marcosgreppi**

### CODE REVIEW

File: 02-oop-inheritance/2-1.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html class="html" lang="">
+  <head>
+    <title>Movie</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <script type="text/javascript" src="2-1.js"> </script>
+    <link rel="stylesheet" href="2-1.css">
+  </head>
+
```
Comment (afiorenza) [inline 02-oop-inheritance/2-1.html]: This break line is not needed

File: 02-oop-inheritance/2-1.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ <body>
+
```
Comment (afiorenza) [inline 02-oop-inheritance/2-1.html]: Same here

File: 02-oop-inheritance/2-1.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  movie1.play();
+
+let elCuerpo = new classMovie ("El Cuerpo", "2012", "106 minutes");
+  console.log(movie2.title);
+  console.log(movie2.year);
+  console.log(movie2.duration);
+  movie2.play();
+
+
+let Social = {
```
Comment (afiorenza) [inline 02-oop-inheritance/2-1.js]: You could move this to another file

File: 02-oop-inheritance/2-1.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import {Logger} from './classLogger';
+import {movieClass} from './classMovie';
+import {Actor} from './classActor';
+
+
+let terminator = new classMovie ("Terminator", "1984", "107 minutes");
+  console.log(movie1.title);
+  console.log(movie1.year);
+  console.log(movie1.duration);
+  movie1.play();
```
Comment (afiorenza) [inline 02-oop-inheritance/2-1.js]: Try to register an event in the event-emiter

File: 02-oop-inheritance/2-1.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+let elCuerpo = new classMovie ("El Cuerpo", "2012", "106 minutes");
+  console.log(movie2.title);
+  console.log(movie2.year);
+  console.log(movie2.duration);
+  movie2.play();
```
Comment (afiorenza) [inline 02-oop-inheritance/2-1.js]: Movie 2 is not defined...

File: 02-oop-inheritance/2-1.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import {eventEmitter} from "./classEventEmitter";
+import {Logger} from './classLogger';
+import {movieClass} from './classMovie';
+import {Actor} from './classActor';
+
+
+let terminator = new classMovie ("Terminator", "1984", "107 minutes");
+  console.log(movie1.title);
```
Comment (afiorenza) [inline 02-oop-inheritance/2-1.js]: How this work as movie1 is not defined?

File: 02-oop-inheritance/2-1.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+let elCuerpo = new classMovie ("El Cuerpo", "2012", "106 minutes");
+  console.log(movie2.title);
+  console.log(movie2.year);
+  console.log(movie2.duration);
+  movie2.play();
+
+
+let Social = {
+  share: function (friendName) {
+    console.log("share", this.title, "with", friendName)
```
Comment (afiorenza) [inline 02-oop-inheritance/2-1.js]: Use es6 template string here

File: 02-oop-inheritance/2-1.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  like: function (friendName) {
+    console.log(friendName, "likes", this.title)
```
Comment (afiorenza) [inline 02-oop-inheritance/2-1.js]: Same here

File: 02-oop-inheritance/classEventEmitter.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  constructor () {
+    this.listeners = []
+  }
+  emit (movie, event) {
+    if (this.listeners[event]) {
+      this.listeners[event](movie, event)
+    }
+  }
+  on (event, callback) {
+    if (this.listeners[event] != callback) {
```
Comment (afiorenza) [inline 02-oop-inheritance/classEventEmitter.js]: Use `!==` as it is more strict

File: 02-oop-inheritance/classLogger.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+class Logger {
+ constructor (){
+ }
+
+ log (movieExecute, functionName){
+     console.log(movieExecute.title + ' ' + functionName)
```
Comment (afiorenza) [inline 02-oop-inheritance/classLogger.js]: Use es6 template strings here

File: 02-oop-inheritance/classMovie.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    super.emit(this, 'play');
+  }
+  pause(){
+    super.emit(this, 'pause');
+  }
+  resume(){
+    super.emit(this, 'resume');
+  }
+  addCast(newActors) {
+    this.actors = this.actors.concat(newActors);
```
Comment (afiorenza) [inline 02-oop-inheritance/classMovie.js]: Initialize this attribute in the class

**Review by afiorenza — CHANGES_REQUESTED** [review]
A few changes. Also next time you should branch from master because here i can see code that is not from this topic. Also where is package.json? You must upload it too.

File: 02-oop-inheritance/2-1.html (hunk 1 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/2-1.html]: removed

File: 02-oop-inheritance/2-1.html (hunk 2 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/2-1.html]: removed

File: 02-oop-inheritance/2-1.js (hunk 6 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/2-1.js]: fixed, changed name at the last minute forgot about the rest

File: 02-oop-inheritance/2-1.js (hunk 5 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/2-1.js]: fixed, changed name at the last minute forgot about the rest

File: 02-oop-inheritance/2-1.js (hunk 3 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/2-1.js]: moved

File: 02-oop-inheritance/2-1.js (hunk 7 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/2-1.js]: fixed

File: 02-oop-inheritance/2-1.js (hunk 8 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/2-1.js]: fixed

File: 02-oop-inheritance/classEventEmitter.js (hunk 9 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/classEventEmitter.js]: fixed

File: 02-oop-inheritance/classLogger.js (hunk 10 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/classLogger.js]: fixed

File: 02-oop-inheritance/classMovie.js (hunk 11 above)
Comment (Marcosgreppi) [reply 02-oop-inheritance/classMovie.js]: changed

**Review by afiorenza — CHANGES_REQUESTED** [review]
Template strings are not well implemented. Check this documentation https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals

**Review by afiorenza — APPROVED** [review]

**Pull request merged**